pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -359,16 +359,76 @@ func TestPriorityQueue_Delete(t *testing.T) {
 }
 
 func TestPriorityQueue_MoveAllToActiveOrBackoffQueue(t *testing.T) {
-	q := NewPriorityQueue(newDefaultQueueSort())
+	c := clock.NewFakeClock(time.Now())
+	q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c))
 	q.Add(&medPriorityPod)
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&unschedulablePod), q.SchedulingCycle())
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&highPriorityPod), q.SchedulingCycle())
+	// Move clock by 1 second so that pods is guaranteed to be added into unschedulableQ first and
+	// then the move request is happened later, otherwise there is potential a test flake here, as
+	// the move request has a chance to be finished before the pods get into the unschedulableQ.
+	// In that case, the pods will be moved to backoffQ even `QueuedPodInfo.Attempts` is zero.
+	c.Step(time.Second)
```

Comment:
Could you rather leave the original test case without the Step?

Or we could have 2 tests.

Reply:
Done, it's here: https://github.com/kubernetes/kubernetes/pull/94413